internal/pipeline/executor.go

```diff
@@ -288,25 +266,42 @@ func hydrate(p *pkg, hctx *hydrationContext) (resources []*yaml.RNode, err error
 			&kio.PackageBuffer{Nodes: input},
 		},
 		Filters: filters,
-		Outputs: []kio.Writer{pkgWriter, output}, // here may be we don't want to write to the fs yet
+		Outputs: []kio.Writer{output},
 	}
 	err = kioPipeline.Execute()
 	if err != nil {
-		err = fmt.Errorf("failed to run pipeline for pkg %s %w", p.path, err)
+		err = fmt.Errorf("failed to run pipeline for pkg %v %w", currPkg.Path(), err)
 		return
 	}
 
 	// pkg is hydrated
-	resources = output.Nodes
 	// mark the pkg as wet and update the resources
-	hctx.hydrationStates[p.Path()] = Wet
-	p.resources = resources
+	resources = output.Nodes
+	currPkg.state = Wet
+	currPkg.resources = resources
+
 	return resources, err
 }
 
-// filterMetaData filters kpt metadata files such as pipeline,
+// adjustRelPath updates the resources with given relative path.
+func adjustRelPath(resources []*yaml.RNode, relPath string) ([]*yaml.RNode, error) {
+	for _, transRes := range resources {
+		meta, err := transRes.GetMeta()
+		if err != nil {
+			return resources, err
+		}
+		newPath := path.Join(relPath, meta.Annotations[kioutil.PathAnnotation])
+		err = transRes.PipeE(yaml.SetAnnotation(kioutil.PathAnnotation, newPath))
+		if err != nil {
+			return resources, err
+		}
+	}
+	return resources, nil
+}
+
+// filterMetaResources filters kpt metadata files such as pipeline,
 // Kptfile, permissions etc.
-func filterMetaData(resources []*yaml.RNode) []*yaml.RNode {
+func filterMetaResources(resources []*yaml.RNode) []*yaml.RNode {
```

Comment:
This should filter out function config specified in`pipeline.config` right? Please have a test for all the cases here.